### PR TITLE
Issues/13 update search bundle

### LINKED ISSUE #13: Update Search Bundle
The search bundle on which the reports are based are published in https://github.com/medizininformatik-initiative/kds-report/blob/gen-search-bundle/dsf-search-bundle.json.

### LINKED ISSUE #14: HRP should be configurable
1. use hrp-identifier provided from task, if not present
2. use hrp-identifier provided from ENV variable, if not present
3. search hrp-identifier for mii-parent-organization and use first found 

Third option is the default option.

### PR DESCRIPTION
**Pull request by wetret**
Updates search bundle to newest version defined by HRP.

Makes HRP configurable by the following rules:

1. use hrp-identifier provided from task, if not present..
2. use hrp-identifier provided from ENV variable, if not present..
3. search hrp-identifier for mii-parent-organization and use first found (default)

resolves #13
resolves #14

### CODE REVIEW

File: src/test/resources/fhir/Bundle/search-bundle.xml [hunk 1]
```diff
@@ -1,976 +1,539 @@
-<Bundle xmlns="http://hl7.org/fhir">
+<Bundle xmlns='http://hl7.org/fhir'>
```
Comment (hhund) [inline src/test/resources/fhir/Bundle/search-bundle.xml:1]: This seems to be the only xml file in the project that uses single quote characters for attributes, consider reverting to double quote characters to stay consistent.

**Review by hhund — APPROVED** [review]
A few minor suggestions here and as line-by-line comments.

In lines [58](https://github.com/medizininformatik-initiative/mii-process-report/blob/36f4e579ec035a7f810f393e1de5ed8387aefeeb/src/main/java/de/medizininformatik_initiative/process/report/service/DownloadSearchBundle.java#L58), [89](https://github.com/medizininformatik-initiative/mii-process-report/blob/36f4e579ec035a7f810f393e1de5ed8387aefeeb/src/main/java/de/medizininformatik_initiative/process/report/service/DownloadSearchBundle.java#L89) and [93](https://github.com/medizininformatik-initiative/mii-process-report/blob/36f4e579ec035a7f810f393e1de5ed8387aefeeb/src/main/java/de/medizininformatik_initiative/process/report/service/DownloadSearchBundle.java#L93) of the `DownloadSearchBundle` class, log messages and an exception message use the phrase "referenced in Task". Consider rephrasing since the search bundle is not referenced in the task, but rather a constant identifier is used to search for the bundle. Also consider rephrasing lines [87](https://github.com/medizininformatik-initiative/mii-process-report/blob/36f4e579ec035a7f810f393e1de5ed8387aefeeb/src/main/java/de/medizininformatik_initiative/process/report/service/CreateReport.java#L87) and [91](https://github.com/medizininformatik-initiative/mii-process-report/blob/36f4e579ec035a7f810f393e1de5ed8387aefeeb/src/main/java/de/medizininformatik_initiative/process/report/service/CreateReport.java#L91) of `CreateReport`.

File: src/main/java/de/medizininformatik_initiative/process/report/service/SelectTargetHrp.java [hunk 2]
```diff
@@ -62,19 +113,21 @@ private Organization getHrpOrganization(Identifier parentIdentifier, Coding role
...
-	private Identifier extractHrpIdentifier(Organization organization)
+	private String extractHrpIdentifierFromOrganization(Organization organization)
 	{
 		return organization.getIdentifier().stream()
 				.filter(i -> NamingSystems.OrganizationIdentifier.SID.equals(i.getSystem())).findFirst()
 				.orElseThrow(() -> new RuntimeException("organization with id '" + organization.getId()
-						+ "' is missing identifier with system '" + NamingSystems.OrganizationIdentifier.SID + "'"));
+						+ "' is missing identifier with system '" + NamingSystems.OrganizationIdentifier.SID + "'"))
+				.getValue();
+	}
```
[the remark this answers is not in the document]
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/process/report/service/SelectTargetHrp.java]: implemented with 178d1fa

File: src/main/java/de/medizininformatik_initiative/process/report/spring/config/ReportConfig.java [hunk 3]
```diff
@@ -34,6 +36,11 @@ public class ReportConfig
 	@Autowired
 	private FhirClientConfig fhirClientConfig;
 
+	@ProcessDocumentation(required = true, processNames = {
```
[the remark this answers is not in the document]
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/process/report/spring/config/ReportConfig.java]: implemented with 178d1fa

File: src/test/resources/fhir/Bundle/search-bundle.xml (hunk 1 above)
Comment (wetret) [reply src/test/resources/fhir/Bundle/search-bundle.xml]: implemented with 178d1fa

**Pull request merged**